cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
> That's incorrect. This is executed in the context of the user's build script, so it checks whether the user has set BUILD_SHARED_LIBS.

True, but I am actually focussing on the build outcome when talking about ```BUILD_SHARED_LIBS```. So what I am getting at is whether dynamic OpenCV libs were built or not. The same goes for the two tables below.

> Wouldn't it find the dynamic build, instead?

No, there aren't any, see the previous paragraph.

The current behaviour (without the patch) is the following:

|                         | OpenCV built with BUILD_SHARED_LIBS=ON | OpenCV built with BUILD_SHARED_LIBS=OFF |
| ----------------------- | -------------------------------------- | --------------------------------------- |
| OpenCV_STATIC=ON        | package found                          | package found                           |
| OpenCV_STATIC=OFF       | package found                          | package NOT found                       |
| OpenCV_STATIC undefined | package found                          | package NOT found                       |

The behaviour with the proposed patch:

|                         | OpenCV built with BUILD_SHARED_LIBS=ON | OpenCV built with BUILD_SHARED_LIBS=OFF |
| ----------------------- | -------------------------------------- | --------------------------------------- |
| OpenCV_STATIC=ON        | package found                          | package found                           |
| OpenCV_STATIC=OFF       | package found                          | package NOT found                       |
| OpenCV_STATIC undefined | package found                          | package  found                          |

The ```OpenCV_STATIC``` field in the tables above refers to the settings prior to calling ```find_package()``` in an example build that wants to use OpenCV like so:
```
set( OpenCV_STATIC ON ) # or OFF or commented out
find_package( OpenCV NO_MODULE )
```